etl-destinations/src/bigquery/core.rs

```diff
@@ -66,13 +68,18 @@ pub fn table_name_to_bigquery_table_id(table_name: &TableName) -> BigQueryTableI
 
 /// Internal state for [`BigQueryDestination`] wrapped in `Arc<Mutex<>>`.
 ///
-/// Contains the BigQuery client, dataset configuration, and injected schema cache.
+/// Contains the BigQuery client, dataset configuration, injected schema cache,
+/// and table creation state cache for performance optimization.
 #[derive(Debug)]
 struct Inner<S> {
     client: BigQueryClient,
     dataset_id: BigQueryDatasetId,
     max_staleness_mins: Option<u16>,
     schema_store: S,
+    /// Cache of table IDs that have been successfully created or verified to exist.
+    /// This avoids redundant `create_table_if_missing` calls for known tables.
+    /// Limited to `MAX_CREATED_TABLES_CACHE_SIZE` entries with random eviction when full.
```

Comment:
For now, added random eviction, but we can have LRU policy if we want to improve the performance.